app/models/application_request.rb

```diff
@@ -40,7 +40,7 @@ def self.req_id(date, req_type, retries = 0)
     req_type_id = req_types[req_type]
 
     # a poor man's upsert
-    id = where(date: date, req_type: req_type_id).pluck_first(:id)
+    id = where(date: date, req_type: req_type_id).pick(:id)
```

Comment:
I wonder, what is better? This + `pick` or `find_or_create_by(...).id`?

Reply:
I looked into this a bit. I think the author might have done this because they explicitly pass `count: 0`, but since that's the default, it shouldn't be needed. And since there's a unique index on `date,req_type`, we can even use `#create_or_find_by` which is immune to race conditions.